app/models/service.js

```diff
@@ -400,6 +401,58 @@ var Service = Resource.extend({
       return Util.formatSi(this.get('launchConfig.memoryReservation'), 1024, 'iB', 'B');
     }
   }),
+
+  localizedServiceUILabel: function(){
+    let labels = this.get('launchConfig.labels');
+    let serviceUILabel = labels[C.LABEL.SERVICE_UI_LABEL] || '';
+    let serviceName = this.get('name');
+    if(!serviceUILabel){
+      return;
+    }
+    let language = this.get('intl._locale')[0];
+    let localizedLabel = '';
+    // if the label is not a valid JSON map, default to use the service name.
+    try{
+      serviceUILabel = JSON.parse(serviceUILabel);
+      // if the label is not a Object, just use service name as label
+      if(serviceUILabel.constructor !== Object){
+        console.warn(`${C.LABEL.SERVICE_UI_LABEL} in ${serviceName} is not a JSON map`);
+        return this.get('name');
+      }
+      localizedLabel = serviceUILabel[Object.keys(serviceUILabel)[0]];
+      if(serviceUILabel[language]){
+        localizedLabel = serviceUILabel[language];
+      }
+    }
+    catch(err){
+      console.warn(`${C.LABEL.SERVICE_UI_LABEL} in ${serviceName} is not valid`);
+      localizedLabel = this.get('name');
+    }
+    return localizedLabel;
+  }.property('launchConfig.labels', 'intl._locale'),
+
+  serviceApp: function(){
+    let labels = this.get('launchConfig.labels');
+    let serviceName = this.get('name');
+    let serviceUIPath = labels[C.LABEL.SERVICE_UI_PATH] || '/';
+    let serviceUIPort = labels[C.LABEL.SERVICE_UI_PORT] || '';
+    let localizedServiceUILabel = this.get('localizedServiceUILabel');
+    let url;
+    if( localizedServiceUILabel ){
+      url = `${window.location.origin}${this.get('app.magicEndpoint')}/projects/${this.get('projectId')}`;
+      if(!serviceUIPort){
```

Comment:
I'll merge it anyway but this is a pet-peeve.. do `if ( condition ) { a } else { b }` instead of `if ( !condition ) { b } else { a }`

Reply:
Sorry, I will be careful next time.